dali/operators/image/crop/crop.h

```diff
@@ -40,10 +40,21 @@ class Crop : public SliceBase<Backend> {
   void ProcessCroppingAttrs(const OpSpec &spec, const workspace_t<Backend> &ws) override {
     const auto &input = ws.template Input<Backend>(0);
     const TensorLayout in_layout = input.GetLayout();
-    DALI_ENFORCE(in_layout.ndim() == input.shape().sample_dim());
-    DALI_ENFORCE(ImageLayoutInfo::HasChannel(in_layout) &&
-      (ImageLayoutInfo::IsImage(in_layout) || VideoLayoutInfo::IsVideo(in_layout)),
-      "Unexpected data layout");
+
+    DALI_ENFORCE(
+        ImageLayoutInfo::HasChannel(in_layout) &&
+            (ImageLayoutInfo::IsImage(in_layout) || VideoLayoutInfo::IsVideo(in_layout)),
+        make_string("Unexpected layout: ", in_layout, "\n\n",
+                    "Expected a non-empty layout including spatial dimensions "
+                    "(H, W, D) and a channel dimension. Examples: \"HWC\", \"DHWC\", \"FHWC\".\n\n"
+                    "To specify a layout manually, use ``fn.reshape``.\n"
+                    "Example:\n\n"
+                    "  ``data = fn.reshape(data, layout=\"HWC\")"));
+
+    DALI_ENFORCE(
```

Comment:
Nitpick: This is checked anyway by the Operator<Backend> (although in Run method so maybe `crop` needs that assured earlier?).

Reply:
There's another PR that actually adds layout info to this op. I think I will merge that one first, then update this one.

Reply:
Sure.